packages/nimble-components/src/utilities/models/tracker.ts

```diff
@@ -0,0 +1,74 @@
+type ObjectFromList<T extends readonly string[]> = {
+    [K in T extends readonly (infer U)[] ? U : never]: boolean;
+};
+
+/**
+ * Generic Tracker which sets or resets provided flags
+ */
+export class Tracker<TrackedItemsList extends readonly string[]> {
+    private trackedItemsState: ObjectFromList<TrackedItemsList>;
+
+    public constructor(trackedItemsList: TrackedItemsList) {
+        type TrackedItems = typeof this.trackedItemsState;
+        this.trackedItemsState = {} as TrackedItems;
+        this.trackedItemsState = trackedItemsList.reduce<TrackedItems>(
+            (r, key): TrackedItems => {
+                return {
+                    ...r,
+                    [key]: false
+                };
+            },
+            this.trackedItemsState
+        );
+    }
+
+    public get trackedItems(): ObjectFromList<TrackedItemsList> {
+        return this.trackedItemsState;
```

Comment:
Instead of returning a reference to the underlying object which could be accidentally mutated by end users can you return a shallow clone?
```suggestion
        return {...this.trackedItemsState};
```

Reply:
When done this should also be changed to a method instead of an accessor, i.e. getTrackedItems()